### PR TITLE
Fix clean QuoteModel, update dependencies, remove some warnings

### PR DESCRIPTION
**Pull request by 4xes**
Maybe I'm wrong with the last commit but for me it helps to run example.

### CODE REVIEW

File: build.gradle [hunk 1]
```diff
@@ -5,7 +5,7 @@ buildscript {
         jcenter()
     }
     dependencies {
-        classpath 'com.android.tools.build:gradle:1.3.0'
+        classpath 'com.android.tools.build:gradle:2.1.0'
```
Comment (thorbenprimke) [inline build.gradle:8]: 👍

File: example/app/build.gradle [hunk 2]
```diff
@@ -31,9 +31,9 @@ android {
 
 dependencies {
     testCompile 'junit:junit:4.12'
-    compile 'com.android.support:support-v4:23.1.1'
-    compile 'com.android.support:appcompat-v7:23.1.1'
-    compile 'com.android.support:recyclerview-v7:23.1.1'
+    compile 'com.android.support:support-v4:23.3.0'
+    compile 'com.android.support:appcompat-v7:23.3.0'
+    compile 'com.android.support:recyclerview-v7:23.3.0'
```
Comment (thorbenprimke) [inline example/app/build.gradle:36]: 👍

File: example/app/src/main/java/co/moonmonkeylabs/realmrecyclerview/example/MainActivity.java [hunk 3]
```diff
@@ -359,15 +357,12 @@ protected Void doInBackground(Void... params) {
...
                 }
                 Realm instance = Realm.getInstance(getRealmConfig());
-                final RealmResults<QuoteModel> all = instance.where(QuoteModel.class).findAll();
-                if (all != null) {
-                    instance.beginTransaction();
-                    all.clear();
-                    instance.commitTransaction();
-                }
+                instance.beginTransaction();
+                instance.delete(QuoteModel.class);
```
Comment (thorbenprimke) [inline example/app/src/main/java/co/moonmonkeylabs/realmrecyclerview/example/MainActivity.java:364]: Good change. Yea, I think this should be fine / I think I read that the functionality of clear on a realmResult list changed. 

👍

File: settings.gradle [hunk 4]
```diff
@@ -1 +1,2 @@
-include ':library'
+include ':library', ':app'
+project(':app').projectDir = new File('example/app')
```
Comment (thorbenprimke) [inline settings.gradle]: This I would prefer not to merge. The example is excluded from the main gradle project on purpose. 
Could you remove it from the PR?

File: library/build.gradle [hunk 5]
```diff
@@ -20,8 +20,10 @@ android {
...
 
 dependencies {
-    compile fileTree(dir: 'libs', include: ['diffutils-1.2.1.jar'])
-    compile 'com.android.support:design:23.1.1'
-    compile 'com.android.support:recyclerview-v7:23.1.1'
-    compile 'com.github.TonicArtos:SuperSLiM:ed0ba4b4d2'
+    compile fileTree(dir: 'libs', include: ['*.jar'])
+    compile 'com.android.support:design:23.3.0'
+    compile 'com.android.support:recyclerview-v7:23.3.0'
+    compile 'com.googlecode.java-diff-utils:diffutils:1.3.0'
```
Comment (thorbenprimke) [inline library/build.gradle:26]: Nice!

File: library/build.gradle [hunk 6]
```diff
@@ -20,8 +20,10 @@ android {
...
+    compile 'com.tonicartos:superslim:0.4.13'
```
Comment (thorbenprimke) [inline library/build.gradle:27]: Nice! Yea, I used the specific commit at the time because Master included a fix I needed.

**Comment by thorbenprimke** [comment]
@4xes - Thanks for the PR. Looks all good except for the `settings.gradle` change. Can you remove it. 

~~For the QuoteModel delete/remove, were you hitting a crash?~~ I saw the `clear` method is depreciated.

**Comment by 4xes** [comment]
Done.

**Pull request merged**